src/audit_simulation_sarah.py

```diff
@@ -0,0 +1,155 @@
+        if type(audit_tier) is not int:
+            raise TypeError("Audit tier must be integer.")
+        if type(invalid) is not bool:
+            raise TypeError("Invalid must be True/False")
+
+        # Set attributes
+        self.total_votes = total_votes
+        self.vote_dist = vote_dist
+        self.invalid = invalid
+        self.audit_tier = audit_tier
+
+        # TODO: audit tier is currently first audit tier, then doubles after
+        #   could modify to accept list of integer audit tiers
+
+        # Get number of candidates in election
+        if invalid:
+            self.num_candidates = len(vote_dist)-1
+        else:
+            self.num_candidates = len(vote_dist)
+
+    def run(self, risk_limit: float) -> bool:
+        """
+        Run (simulate) Bayesian RLA.
+
+        @param risk_limit: float value (between 0 and 1) to use in audit.
+
+        @return: Returns True is audit stops, i.e. confirming anounced outcome
+        of election. False if audit must progress to a full recount.
+
+        Testing
+
+        >>> auditsim(10000, [9000, 1000], 100).run(0.1)
+        True
+        >>> auditsim(10000, [5000, 5000], 100).run(0.1)
+        False
+        >>> auditsim(10000, [4000, 6000], 100).run(0.1)
+        False
+        """
+        # Type check risk limit
+        if type(risk_limit) is not float:
+            raise TypeError("Risk limit must be float")
+        if risk_limit >= 1.0 or risk_limit <= 0.0:
+            raise ValueError("Risk limit must be between 0 and 1.0")
+
+        # Create distribution object to sample from ballots in election
+        ballots = multivhyper(self.vote_dist)
+
+        # Initialize array to hold audit counts for each type of vote
+        audit_counts = np.zeros(len(self.vote_dist), dtype=int)
+
+        # Set initial audit tier and sample size
+        audit_tier = self.audit_tier
+        sample_size = self.audit_tier
+
+        # Create auditing object for calculations
+        audit_lookup = brla(self.total_votes)
+
+        # Run audit until stop or full recount
+        while True:
+            # Get a sample of ballots
+            sample = ballots.sample(sample_size)
+
+            # Add ballots in new sample to running count of votes in audit
+            audit_counts = audit_counts + sample
+
+            # Handle invalid votes if necessary
+            if self.invalid:
+                invalid = int(audit_counts[-1])
+            else:
+                invalid = 0
+
+            # Audit comparisons
+            if self.num_candidates == 2:
+                # Two candidate comparison
+                kmin = audit_lookup.get_stopping_size(audit_tier, risk_limit, invalid)
+
+                if audit_counts[0] > kmin:
+                    # print("Audit stopped at size ", audit_tier)
+                    return True
+            else:
+                # Multicandidate comparison
+                stop = True
```

Comment:
might make sense to pull this "stop" logic out to drive the loop itself so it applies to the two-candidate case also.